### PR TITLE
Adding fluent validation for api contracts

### PR DESCRIPTION
**Pull request by tarrball**
This PR adds fluent validation and enforces validation on input that goes through `ExecuteAsync` in the manager.

### CODE REVIEW

File: source/WesternStatesWater.WaDE.Managers.Tests/WaterResourceManager/WaterResourceManagerTestsBase.cs [hunk 1]
```diff
@@ -24,13 +26,20 @@ public abstract class WaterResourceManagerTestsBase
 
     internal Api.WaterResourceManager CreateWaterResourceManager()
     {
+        var logger = new ServiceCollection()
+            .AddLogging(config => config.AddConsole())
+            .BuildServiceProvider()
+            .GetRequiredService<ILoggerFactory>()
+            .CreateLogger<Api.WaterResourceManager>();
```
Comment (zlannin) [inline source/WesternStatesWater.WaDE.Managers.Tests/WaterResourceManager/WaterResourceManagerTestsBase.cs:33]: Why not use `NullLogger.Instance<Api.WaterResourceManager>` unless we are testing the log messages?

**Review by zlannin — APPROVED** [review]
I see a lot of files that added an import but didn't use it in the WaDEApiFunctions/v1 files.

File: source/WesternStatesWater.WaDE.Managers.Tests/WaterResourceManager/WaterResourceManagerTestsBase.cs (hunk 1 above)
Comment (tarrball) [reply source/WesternStatesWater.WaDE.Managers.Tests/WaterResourceManager/WaterResourceManagerTestsBase.cs:33]: It is nice to see logs in console output in tests when exceptions are caught/logged/not re-thrown, but that's probably the only benefit. I can't seem to get your suggestion to work. I have these packages imported:

image

Comment (tarrball) [reply source/WesternStatesWater.WaDE.Managers.Tests/WaterResourceManager/WaterResourceManagerTestsBase.cs:33]: image

**Comment by tarrball** [comment]
> I see a lot of files that added an import but didn't use it in the WaDEApiFunctions/v1 files.

This is because those files were creating errors and the error class's namespace changed. They're still creating the same errors, but had to add an import.

**Pull request merged**